test/e2e/framework/util.go

```diff
@@ -349,6 +432,51 @@ func GetReadySchedulableNodesOrDie(c clientset.Interface) (nodes *v1.NodeList) {
 	return nodes
 }
 
+// newSecretTemplate returns the default v1.Secret template for this jig, but
+// does not actually create the Secret.
+func newSecretData(ns, generateName string, caCert, cert, privateKey, passphrase []byte) *v1.Secret {
```

Comment:
I'd probably in-line all of this into `CreateSSLSecretOrFail` which once the poll is gone won't be too big 🙂